src/main.ts

```diff
@@ -545,6 +546,30 @@ export async function getToken(input: CspInput): Promise<string> {
   }
 }
 
+export async function checkTokenExpiration(input: CspInput): Promise<string> {
+  const response = await cspClient.post("https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/details", {
+    headers: {
+      "Content-Type": "application/json",
+      tokenValue: "$CSP_API_TOKEN",
+    },
+  })
+
+  cachedCspToken = {
+    access_token: response.data.access_token,
+    timestamp: Date.now() + input.timeout,
+  }
+
+  if (response.data.details) {
```

Comment:
This logic looks wrong. It should be placed after the check that you have below.